### ISSUE TITLE
Glide Join OOMs on long source clips: _load_frames decodes the entire video into RAM

### ISSUE THREAD

**Issue by xiaohongai** [CGlide/ComfyUI-CGlide#23]
```
### Summary

`CSGlideJoinCS` runs out of system memory (OOM-killed by the kernel) when
`source_video` is a long clip. The cause is `_load_frames()` in
`csglide_join.py`, which unconditionally decodes the **entire** source video
into a Python list and then `np.stack`s it into a float32 tensor, regardless
of how many frames the join actually needs.

### Environment

- ComfyUI: <version>
- CGlide: <commit hash, e.g. from `git rev-parse HEAD`>
- OS: Linux (Ubuntu <version>)
- RAM: 124 GB
- GPU: <model>
- Workflow: H3 Studio ref2va + Glide Join (early_cut) + Glide Video
- Output canvas: 1664x960, 243 frames @ 24 fps

### Reproduction

1. Render a continuation with `CSGlideCastCS` where `cont.file` points to a
   long clip (in my case ~46 s, ~1100 frames at 24 fps).
2. Wire its `source_video` output into `CSGlideJoinCS`.
3. Run. The join reaches `[Glide Join] level tracked over 22 frames` and is
   then OOM-killed.

`dmesg` confirms:

    oom-kill:constraint=CONSTRAINT_NONE,...,task=python3,pid=...
    HangWatcher invoked oom-killer: gfp_mask=0x140cca(...GFP_HIGHUSER_MOVABLE...), order=0

`order=0` means a single 4 KB page allocation failed — physical RAM was
genuinely exhausted, not a cgroup limit.

### Analysis

`_load_frames(path)` in `csglide_join.py`:

```python
def _load_frames(path):
    frames = []
    depth = 65535.0
    with av.open(path) as container:
        for frame in container.decode(video=0):
            frames.append(frame.to_ndarray(format="rgb48le"))
    ...
    return torch.from_numpy(np.stack(frames).astype(np.float32) / depth)
```

No max_frames, no seek, no trim. For a 46 s clip at 1664x960:

~1100 frames x 9.2 MB/frame (rgb48le) ~= 10 GB in the list

np.stack copies -> +10 GB

.astype(np.float32) -> +20 GB

peak well over 40 GB before ext and the affine fit are even counted

At 37 s (~888 frames) the peak lands around 35-38 GB and squeaks under the
49 GB available; at 46 s it crosses and the OOM killer fires. The threshold
is purely a function of source-clip length, not of overlap_frames or the
output length.

Why this matters
The join only needs the tail of the source for two of its three uses:

_tracked_affine(src, ext, ov) only reads src[-ov:]

early_cut does head = src[:-ov] then torch.cat([head, ext])

hard_cut does torch.cat([src, ext[ov:]]) and does need all of src

So for early_cut (the default, and the mode recommended for H3), the
head of the source is copied verbatim into the output. It never needs to be
in Python memory at all — it could be streamed straight from the source
file into the encoder.

Suggested fix
Option 1 — add an optional cap to _load_frames:
```

def _load_frames(path, max_frames=None):
    frames = collections.deque(maxlen=max_frames) if max_frames else []
    ...
```

and in join(), pass max_frames=ov + ALIGN_SEARCH + 1 for early_cut
and early_scurve. hard_cut still needs the full decode, so gate it on
the mode.

Option 2 — stream the head. For early_cut, skip decoding the source's
head entirely and let Glide Video (or ffmpeg) copy it from the source file
with -c copy. That removes the peak altogether, at the cost of a larger
change to _join_images and the IMAGE interface.

Either way, the tail used for _tracked_affine must stay at rgb48le so
the colour fit keeps its 16-bit input — that part should not change.

Workaround I'm using
Pre-trimming the source to just the overlap window is not an option for me,
because Glide Join keeps src[:-ov] in the output — trimming it would drop
the head of the film. Right now I'm relying on swap to absorb the peak.

Happy to test a patch if you'd like.

```
[Glide Join] level tracked over 22 frames, gain 0.892..0.948, held at 0.925 0.932 0.929 / 0.0217 0.0216 0.0199
Killed

```

**Comment by CGlide** [comment]
Thanks for the write-up — the order=0 detail and the arithmetic made this easy to find, that's a better report than most.

You were right about the four copies. Fixed in the next release: _load_frames now reads the frame count from the container, allocates one float32 array and decodes straight into it. No list, no np.stack, no separate .astype, and the divide is in place. Output is bit-identical to before — I checked the 16-bit path, the 8-bit fallback and the regrow path, max difference 0.0.

That's roughly 3× less peak RAM. Your 46s case goes from ~50 GB to ~21 GB.

One thing on your Option 1: capping the decode at ov + ALIGN_SEARCH + 1 would break early_cut, since that copies src[:-ov] into the output — you'd lose the head of the film, which is the thing you said you couldn't afford. So I left that out.

Option 2 is the right call for the remaining ceiling, and I'm not doing it yet. The 21 GB that's left is the output IMAGE tensor itself, so removing it means the head never becomes an IMAGE, which changes the contract between Glide Join and Glide Video. Also, cutting the source without re-encoding only works on a keyframe, and the cut lands at an arbitrary frame — on H.264 that means either re-encoding the head (a generation of loss on every link in the chain, which is exactly what the rgb48le decode exists to avoid) or mandating all-intra intermediates. Neither is a good trade for most people.

Worth knowing for your workflow in the meantime: the memory scales with the source length, not the overlap or the new clip. If you point CONTINUE FROM at your last clip rather than the assembled film, and join the film together at the end, you never get near this. I've added a note to the tooltip.
